src/averageAges.js

```diff
@@ -55,7 +66,28 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const mothers = people.filter(person => people.some(human =>
+    human.mother === person.name));
+
+  const children = onlyWithSon ? people.filter(person =>
+    person.mother !== null && person.sex === 'm')
```

Comment:
I think this check for null is redundant, bacause you check later if `mother.name === child.mother`. You can try to do this without this check